lib/remap-lang/src/parser.rs

```diff
@@ -0,0 +1,331 @@
+#![allow(clippy::or_fun_call)]
```

Comment:
Note that a lot in this file is copy/pasted from the existing implementation, with the necessary changes to return expressions where relevant and some clean-up to make the parser simpler.